main.go

```diff
@@ -212,17 +219,32 @@ func (m *MetricMap) Register() error {
 	return nil
 }
 
-func (m *MetricMap) UpdateMetricValue(metricName, timespan, repo string, value float64) error {
+func (m *MetricMap) UpdateMetricValue(metricName, timespan, repo string, value float64, labels []MetricLabel) error {
+
+	stringMap := make(map[string]string)
+	stringMap[intervalLabel] = timespan
+	stringMap[repoLabel] = repo
+	for _, l := range labels {
+		stringMap[l.Key] = l.Value
+	}
+
 	gauge := m.Gauges[metricName]
-	gauge.WithLabelValues(timespan, repo).Set(value)
+	gauge.With(stringMap).Set(value)
 	return nil
 }
 
-func (m *MetricMap) AddGauge(metricName, repo, interval string) error {
+func (m *MetricMap) AddGauge(metricName string, labels []MetricLabel) error {
+	var labelSlice []string
```

Comment:
Same thing with the naming here. `labelSlice` tells us little about what difference it is to `labels` parameter.

I will again propose renaming the parameter to `staticLabels` and this value to `labelKeys`.